src/core/experimental/memutils.d

```diff
@@ -0,0 +1,332 @@
+/**
+ * Pure D replacement of the C Standard Library basic memory building blocks of string.h
+ * Source: $(DRUNTIMESRC core/experimental/memutils.d)
+ */
+module core.experimental.memutils;
+
+unittest
```

Comment:
At the top level a user `unittest` should be provided which shows off the core features.
Example: https://dlang.org/phobos/std_container_binaryheap.html

tl;dr: All of these tests should be moved below the declaration of the function.